marshmallow/marshalling.py

```diff
@@ -275,6 +283,22 @@ def deserialize(self, data, fields_dict, many=False, partial=False,
             if value is not missing:
                 key = fields_dict[attr_name].attribute or attr_name
                 set_value(ret, key, value)
+
+        if unknown != IGNORE:
+            for key in set(data) - set(fields_dict):
+                value = data[key]
+                if unknown == ALLOW:
+                    set_value(ret, key, value)
+                elif unknown == RAISE:
+                    def _raise(_):
+                        raise ValidationError('Unkown key.')
+                    self.call_and_store(
+                        getter_func=_raise,
+                        data=value,
+                        field_name=key,
+                        field_obj=type(value)
```

Comment:
`field_obj` is supposed to be the `Field` instance associated with the invalid key. In this case, there isn't a `field_obj`, so `call_and_store` should probably be modified to allow `field_obj` to be omitted.